src/noit_filters.c

```diff
@@ -775,8 +775,8 @@ noit_apply_filterset(const char *filterset,
   const char *local_metric_name = noit_metric_get_full_metric_name(metric);
   mtev_gettimeofday(&now, NULL);
 
-  noit_metric_tag_t *stags = malloc((MAX_TAGS + 1) * sizeof(noit_metric_tag_t));
-  noit_metric_tag_t *mtags = malloc(MAX_TAGS * sizeof(noit_metric_tag_t));
+  noit_metric_tag_t *stags = calloc((MAX_TAGS + 1), sizeof(noit_metric_tag_t));
+  noit_metric_tag_t *mtags = calloc(MAX_TAGS, sizeof(noit_metric_tag_t));
```

Comment:
I don't know why I did this in the first place. we know the size at compile time, so we should just use 
```
noit_metric_tag_t stags[MAX_TAGS + 1], mtags[MAX_TAGS];
```

Reply:
Oh, I did this because, to be compatible with `noit_metric_tagset_builder_end` it needs to be able to `realloc`.

Reply:
>we can call noit_metric_tagset_init to properly set the tagset members to 0, but we still need to manually change the members afterwards.

Why is this not using a builder then?  Looking at the API, it's clear ``noit_metric_tagset_t`` is meant to be initialized from a builder and building tagets, like you're doing now, should use ``noit_metric_tagset_builder_t``

> tldr; there are a few functions at play that don't work very well together and I don't think we should prioritize refactor it right now.

This points to using the builder facility

Reply:
I agree that this should use the builder, but it's not immediately obvious how much refactoring is needed to update `noit_metric_parse_tags` to use the builder since this is not the only place `noit_metric_parse_tags` is used. 
If we want to refactor this, I think we should get the team's approval first since this is a little outside the scope of the ticket and I have other work lined up after implicit tags that may be higher priority.

Reply:
> I agree that this should use the builder, but it's not immediately obvious how much refactoring is needed to update

ok, nothing needed then, if the API is not meant to be used like this then that's fine

> a little outside the scope of the ticket

We would just make a ticket that blocks this one

Reply:
> We would just make a ticket that blocks this one
maybe related, but the builder functions already weren't being used here prior to the implicit tag changes, so this isn't anything that would block this ticket.